lib/src/http/base_request.dart

```diff
@@ -44,7 +44,7 @@ abstract class BaseRequest implements FluriMixin, RequestDispatchers {
   /// Information about this request related to retries is also available here.
   /// This includes the current number of attempts and the current list of
   /// failures.
-  RequestAutoRetry? autoRetry;
+  late RequestAutoRetry autoRetry;
```

Comment:
I did try that at first, but unfortunately we use the setter in several places (like cloning and in the various request mixins), so it wasn't feasible to get rid of the setter. I do wonder if we could use abstract getter and setter here rather than using a field.. the problem with a non-nullable field even on an abstract class is that Dart wants you to initialize it.

Reply:
Tried it out locally and getter + setter does seem to work in place of a `late` field.